api/src/main/java/org/openmrs/module/xdssender/api/notificationspullpoint/impl/NotificationsPullPointClientImpl.java

```diff
@@ -0,0 +1,79 @@
+package org.openmrs.module.xdssender.api.notificationspullpoint.impl;
+
+import ca.uhn.hl7v2.model.Message;
+import org.apache.commons.codec.binary.Base64;
+import org.openmrs.module.xdssender.XdsSenderConfig;
+import org.openmrs.module.xdssender.api.notificationspullpoint.NotificationsPullPointClient;
+import org.openmrs.module.xdssender.notificationspullpoint.GetMessages;
+import org.openmrs.module.xdssender.notificationspullpoint.GetMessagesResponse;
+import org.openmrs.module.xdssender.notificationspullpoint.NotificationMessageHolderType;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import org.springframework.ws.WebServiceMessage;
+import org.springframework.ws.client.core.WebServiceMessageCallback;
+import org.springframework.ws.client.core.support.WebServiceGatewaySupport;
+import org.springframework.ws.transport.context.TransportContext;
+import org.springframework.ws.transport.context.TransportContextHolder;
+import org.springframework.ws.transport.http.HttpUrlConnection;
+
+import javax.xml.transform.TransformerException;
+import java.io.IOException;
+import java.math.BigInteger;
+import java.nio.charset.Charset;
+import java.util.ArrayList;
+import java.util.List;
+
+@Component("xdssender.NotificationsPullPointClientImpl")
+public class NotificationsPullPointClientImpl extends WebServiceGatewaySupport implements NotificationsPullPointClient {
+	
+	private BigInteger MAX_MESSAGES_PER_REQUEST = BigInteger.valueOf(100);
```

Comment:
shouldn't it be a property?

Reply:
if not -> static final?

Reply:
Yes, maybe it's good idea for it to be a property